client/app/competitor/QuestionNavbar.tsx

```diff
@@ -0,0 +1,150 @@
+
+export default function QuestionNavbar({
+    setCurrentQuestion,
+}: {
+    setCurrentQuestion: (question: Question) => void;
+}) {
+    const [questions] = useState<Question[]>([
+        {
+            question: 'Sort an Array of Integers',
+            description: 'Sort an array of integers in ascending order and return it.',
+            tests: [
+                {
+                    input: '2 11 15 0',
+                    output: '0 2 11 15',
+                    failedOutput: '2 0 11 15',
+                    expectedOutput: '0 2 11 15',
+                },
+            ],
+            status: 'complete',
+        },
+        {
+            question: 'Sort an Array of Characters Alphabetically',
+            description:
+                'Sort an array of characters alphabetically and return them as a single string.',
+            tests: [
+                {
+                    input: 'h e l o',
+                    output: 'e h l o',
+                    failedOutput: 'h o e l',
+                    expectedOutput: 'e h l o',
+                },
+            ],
+            status: 'in-progress',
+        },
+        {
+            question: 'Hexadecimal in Reverse Order',
+            description:
+                'Convert characters to hexadecimal values and return them in reverse order.',
+            tests: [
+                {
+                    input: 'A B C D',
+                    output: '13 12 11 10',
+                    failedOutput: '10 11 12 13',
+                    expectedOutput: '13 12 11 10',
+                },
+            ],
+            status: 'failed',
+        },
+        {
+            question: 'Some of Digits',
+            description:
+                'Write a function that takes a positive integer as input and returns the sum of its digits.',
+            tests: [
+                {
+                    input: '1 2 3 4',
+                    output: '10',
+                    failedOutput: '5',
+                    expectedOutput: '10',
+                },
+            ],
+            status: 'not-attempted',
+        },
+    ]);
+
+    const [selectedQuestion, setSelectedQuestion] = useState<Question>(questions[0]);
+
+    const questionStatusColor = (questionState: QuestionState) => {
+        const classMap: Record<QuestionState, string> = {
+            complete: 'border-pass',
+            'in-progress': 'border-in-progress',
+            'not-attempted': 'border-not-attempted',
+            failed: 'border-fail ',
+        };
+        return classMap[questionState];
+    };
+
+    const selectedQuestionColor = (questionState: QuestionState) => {
+        const classMap: Record<QuestionState, string> = {
+            complete: 'bg-pass',
+            'in-progress': 'bg-in-progress',
+            'not-attempted': 'bg-not-attempted',
+            failed: 'bg-fail ',
+        };
+        return classMap[questionState];
+    };
```

Comment:
> Two things:
> 
> * These should use the variables that we define in the tailwind config
> 
> * This should use a lookup table instead of a switch/case.  Look at how it is done in the leaderboard.

This changed the colour of the active question highlight to be solid.  Try adjusting the opacity to something like `30%` with `bg-pass/30`, etc.

See: https://tailwindcss.com/docs/colors#adjusting-opacity